src/components/tl/ucp/reduce/reduce_knomial.c

```diff
@@ -18,27 +18,37 @@
 
 ucc_status_t ucc_tl_ucp_reduce_knomial_progress(ucc_coll_task_t *coll_task)
 {
-    ucc_tl_ucp_task_t *task = ucc_derived_of(coll_task, ucc_tl_ucp_task_t);
-    ucc_coll_args_t   *args = &coll_task->args;
-    ucc_tl_ucp_team_t *team = TASK_TEAM(task);
+    ucc_tl_ucp_task_t *task      = ucc_derived_of(coll_task, ucc_tl_ucp_task_t);
+    ucc_coll_args_t   *args      = &coll_task->args;
+    ucc_tl_ucp_team_t *team      = TASK_TEAM(task);
     ucc_rank_t         myrank    = team->rank;
     ucc_rank_t         team_size = team->size;
     ucc_rank_t         root      = (ucc_rank_t)args->root;
     uint32_t           radix     = task->reduce_kn.radix;
     ucc_rank_t         vrank     = (myrank - root + team_size) % team_size;
     void              *rbuf      = (myrank == root) ? args->dst.info.buffer :
                                                       task->reduce_kn.scratch;
-    ucc_memory_type_t  mtype     = args->src.info.mem_type;
-    ucc_datatype_t     dt        = args->src.info.datatype;
-    size_t             count     = args->src.info.count;
-    size_t             data_size = count * ucc_dt_size(dt);
-    void              *received_vectors =
-                           PTR_OFFSET(task->reduce_kn.scratch, data_size);
-    void              *scratch_offset;
+    ucc_memory_type_t  mtype;
+    ucc_datatype_t     dt;
+    size_t             count, data_size;
+    void              *received_vectors, *scratch_offset;
     ucc_rank_t         vpeer, peer, vroot_at_level, root_at_level, pos;
     uint32_t           i;
     ucc_status_t       status;
 
+    if (root == myrank) {
```

Comment:
same as above

Reply:
if use "?" instead of if else i will need to write same ternary operator thing 4 times and here it's just 1 if